java/image.bzl

```diff
@@ -114,19 +114,20 @@ def java_files(f):
     Returns:
         Depset of java source files.
     """
-    files = depset()
+    files = []
+
     if java_common.provider in f:
         java_provider = f[java_common.provider]
-        files = depset(transitive = [files, java_provider.transitive_runtime_jars])
-    if hasattr(f, "files"):  # a jar file
-        files = depset(transitive = [files, f.files])
-    return files
+        files.append(java_provider.transitive_runtime_jars)
+
+    f_files = f[DefaultInfo].files
+    if f_files != None:
+        files.append(f_files)
+
+    return depset(transitive = files)
 
 def java_files_with_data(f):
-    files = java_files(f)
-    if hasattr(f, "data_runfiles"):
-        files = depset(transitive = [files, f.data_runfiles.files])
-    return files
+    return depset(transitive = [java_files(f), f[DefaultInfo].data_runfiles.files])
```

Comment:
why not checking for f[DefaultInfo].data_runfiles.files not None here?

Reply:
There are many other places where this needs to be checked. I thought I'll do it in a separate commit. I've also left a comment https://github.com/bazelbuild/bazel/issues/3015#issuecomment-471513538 to see if `None` can be removed so that we don't need to have these checks.

Reply:
I think let's still do a check of `f[DefaultInfo].data_runfiles` being not `None` in this PR to leaving this at a safe state.

Reply:
I can do that, np. But there are many other places in code where this check is not performed. Do you want me to fix them too?

Reply:
That would be very appreciated! But they don't have to be in this PR.